modules/ossm-installing-gateway-using-gateway-injection.adoc

```diff
@@ -0,0 +1,246 @@
+----
+
+. Create a YAML file named `gateway-service.yml` that contains the Kubernetes `Service` for the gateway. 
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  type: ClusterIP <1>
+  selector:
+    istio: <gateway_name> <2>
+  ports:
+    - name: status-port
+      port: 15021
+      protocol: TCP
+      targetPort: 15021
+    - name: http2
+      port: 80
+      protocol: TCP
+      targetPort: 80
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: 443
+----
+<1> When you  set `spec.type` to `ClusterIP` the gateway `Service` is available to traffic within the cluster. If the gateway handles ingress traffic from outside the cluster, set `spec.type` to `LoadBalancer`. Alternatively, you can use OpenShift Routes. 
+<2> Set the `selector` to the unique label or set of labels specified in the pod template of the gateway `Deployment` that you previously created.
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-service.yml
+----
+
+. Verify that the gateway `Service` is targeting the `Endpoints` of gateway pods by running the following command:
++
+[source,terminal]
+----
+$ oc get endpoints <gateway_name> -n <gateway_namespace>
+----
++
+You should see output similar to the following example:
++
+.Example output
+[source,terminal]
+----
+NAME              ENDPOINTS                             AGE
+<gateway_name>    10.131.0.181:8080,10.131.0.181:8443   1m
+----
+
+. Optional: Create a YAML fle named `secret-reader.yml` that defines a `HorizontalPodAutoscaler` for the gateway. The following example sets the minimum replicas to `2` and the maximum replicas to `5` and scales the replicas up when average CPU utilization exceeds 80% of the CPU resource limit. This limit is specified in the pod template of the `Deployment` for the gateway.  
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  minReplicas: 2
+  maxReplicas: 5
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 80
+        type: Utilization
+    type: Resource
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: <gateway_name> <1>
+----
+<1> Set `spec.scaleTargetRef.name` to the name of the gateway `Deployment` previously created.
+
+. Optional: Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-hpa.yml
+----
+
+. Optional: Create a YAML file named `gateway-pdb` that defines a `PodDisruptionBudget` for gateway. This example allows gateway pods to be evicted only when at least 1 healthy gateway pod will remain on the cluster after the eviction. The following example uses `gateway-pdb.yml` as the name of the file.
```

Comment:
```suggestion
. Optional: Create a YAML file named `gateway-pdb.yml` that defines a `PodDisruptionBudget` for the gateway. This example allows gateway pods to be evicted only when at least 1 healthy gateway pod will remain on the cluster after the eviction.
```

Reply:
done

Reply:
I think we can also delete the last sentence:

> The following example uses `gateway-pdb.yml` as the name of the file.

Reply:
done